mifosng-android/src/main/java/com/mifos/mifosxdroid/online/collectionsheetindividual/IndividualCollectionSheetFragment.java

```diff
@@ -0,0 +1,413 @@
+
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    void setRepaymentDate() {
+        mDatePicker = MFDatePicker.newInsance(this);
+        String date = DateHelper.
+                getDateAsStringUsedForCollectionSheetPayload(MFDatePicker.getDatePickedAsString());
+        tvRepaymentDate.setText(date.replace('-', ' '));
+        transactionDate = date.replace('-', ' ');
+        actualDisbursementDate = transactionDate;
+    }
+
+    private void prepareRequestPayload() {
+        requestPayload = new RequestCollectionSheetPayload();
+        requestPayload.setOfficeId(mOfficeId);
+        requestPayload.setStaffId(mStaffId);
+        requestPayload.setTransactionDate(tvRepaymentDate.getText().toString());
+
+
+        /* Below 3 lines for Testing only */
+
+        /*requestPayload.setOfficeId(1);
+        requestPayload.setStaffId(12);
+        requestPayload.setTransactionDate("06 July 2017");*/
+    }
+
+    private void setUpUi() {
+        setRepaymentDate();
+        mOfficeNameList = new ArrayList<>();
+        mOfficeAdapter = new ArrayAdapter<>(getActivity(),
+                android.R.layout.simple_spinner_item, mOfficeNameList);
+        mOfficeAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        spOffices.setAdapter(mOfficeAdapter);
+        spOffices.setOnItemSelectedListener(this);
+
+        mStaffNameList = new ArrayList<>();
+        mStaffAdapter = new ArrayAdapter<>(getActivity(),
+                android.R.layout.simple_spinner_item, mStaffNameList);
+        mStaffAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        spStaff.setAdapter(mStaffAdapter);
+
+        if (mSheet != null) {
+            showCollectionSheetViews(mSheet);
+        }
+
+        tvRepaymentDate.setOnClickListener(this);
+        btnFetchSheet.setOnClickListener(this);
+        presenter.fetchOffices();
+
+    }
+
+    @Override
+    public void onShowSheetMandatoryItem(BulkRepaymentTransactions transaction, int position) {
+        //Add the data as received
+        mPayload.getBulkRepaymentTransactions().set(position, transaction);
+    }
+
+    @Override
+    public void onSaveAdditionalItem(BulkRepaymentTransactions transaction, int position) {
+        //Add or amend the data as changed by the user
+        mPayload.getBulkRepaymentTransactions().set(position, transaction);
+    }
+
+
+    private void showCollectionSheetViews(IndividualCollectionSheet sheet) {
+
+        if (tvNoSheetFound.getVisibility() == View.VISIBLE) {
+            tvNoSheetFound.setVisibility(View.GONE);
+        }
+
+        paymentTypeList = presenter.filterPaymentTypeOptions(sheet.getPaymentTypeOptions());
+        loansAndClientNames = presenter.filterLoanAndClientNames(sheet.getClients());
+
+        //Initialize payload's BulkRepaymentTransactions array with dummy values
+        //so that we change the item's content later at some position
```

Comment:
It should have been _default values_ instead of _dummy values_. I'll change it to convey in a better way.